assets/js/phoenix_live_view/rendered.js

```diff
@@ -17,6 +19,90 @@ import {
   isCid,
 } from "./utils"
 
+const VOID_TAGS = [
+  "area",
+  "base",
+  "br",
+  "col",
+  "command",
+  "embed",
+  "hr",
+  "img",
+  "input",
+  "keygen",
+  "link",
+  "meta",
+  "param",
+  "source",
+  "track",
+  "wbr"
+]
+
+export let modifyRoot = (html, attrs, clearInnerHTML) => {
+  let i =0
+  let insideComment = false
+  let insideTag = false
+  let tag
+  let beforeTagBuff = []
+  while(i < html.length){
+    let char = html.charAt(i)
+    if(insideComment){
+      if(char === "-" && html.slice(i, i + 3) === "-->"){
+        insideComment = false
+        beforeTagBuff.push("-->")
+        i += 3
+      } else {
+        beforeTagBuff.push(char)
+        i++
+      }
+    } else if(char === "<" && html.slice(i, i + 4) === "<!--"){
+      insideComment = true
+      beforeTagBuff.push("<!--")
+      i += 4
+    } else if(char === "<"){
+      insideTag = true
+      let iAtOpen = i
+      for(i; i < html.length; i++){
+        if([">", " ", "\n", "\t", "\r"].indexOf(html.charAt(i)) >= 0){ break }
```

Comment:
Here's another place where I suspect you can optimize the inclusion lookups (`indexOf`), and this particular line could pay great dividends to optimize due to being executed within the `while` and `for` loops.

Reply:
set is a good call out to use for these, but it won't likely be noticeable in the hot path where we are using this. The entire Rendered structure to string is usually 1ms or less, including this, while patching is on the order of 1-tens of milliseconds.